ecommerce/CheckoutController.php

```diff
@@ -836,6 +826,38 @@ private function set_pay_now_alert_msg( $order_data ) {
 		}
 	}
 
+	/**
+	 * Pay for the incomplete order
+	 *
+	 * Redirect to the payment gateway to complete the order
+	 * After completing the process it will redirect user to
+	 * order placement page
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return void
+	 */
+	public function pay_incomplete_order() {
+		$order_id = Input::get( 'order_id', 0, Input::TYPE_INT );
```

Comment:
```
if ( 'tutor_pay_incomplete_order' !== Input::post( 'tutor_action' ) {
  return;
}
```

Nonce check missing for tutor_action hook callback
`tutor_utils()->checking_nonce();`